src/error.rs

```diff
@@ -50,6 +50,8 @@ pub enum DecodeError {
     NotCompact,
     #[error("missing type byte")]
     MissingType,
+    #[error("only Helium Ed25519 keys may be converted to a Solana key")]
+    UnsupportedSolanaConversion,
```

Comment:
Can we make this more generic? i.e. `Unsupported(String)` and pass that message in as part of the TryFrom? There'll likely be other unsupported decode error calls

Reply:
is `&'static str` ok?